pdns/auth-domaincache.hh

```diff
@@ -0,0 +1,96 @@
+/*
+ * This file is part of PowerDNS or dnsdist.
+ * Copyright -- PowerDNS.COM B.V. and its contributors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * In addition, for the avoidance of any doubt, permission is granted to
+ * link this program with OpenSSL and to (re)distribute the binaries
+ * produced as the result of such linking.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#pragma once
+#include <string>
+#include <unordered_map>
+#include "dnsname.hh"
+#include "lock.hh"
+
+class AuthDomainCache : public boost::noncopyable
+{
+public:
+  AuthDomainCache(size_t mapsCount = 1024);
+  ~AuthDomainCache();
+
+  void replace(const vector<tuple<DNSName, int>>& domains);
+  void add(const DNSName& domain, const int zoneId);
+  void setReplacePending();  //!< call this when data collection for subsequent replace() call starts.
```

Comment:
I can't parse this at 1am, maybe it makes sense at some other hour

Reply:
I have no trouble with it, and I do not have better words